libs/src/sanityChecks/rolloverNodes.js

```diff
@@ -1,69 +1,88 @@
+const _ = require('lodash')
+
 const Utils = require('../utils')
 const CreatorNode = require('../services/creatorNode')
 
 const THREE_SECONDS = 3000
 const MAX_TRIES = 3
 
-/** Check if the user's primary creator node is healthy */
-const checkPrimaryHealthy = async (libs, primary, tries) => {
-  const healthy = await Utils.isHealthy(primary)
+const nodeIsHealthy = async (endpoint, tries) => {
+  const healthy = await Utils.isHealthy(endpoint)
   if (healthy) return healthy
   else {
-    if (tries === 0) {
-      return false
-    }
+    if (tries === 0) return false
     await Utils.wait(THREE_SECONDS)
-    return checkPrimaryHealthy(libs, primary, tries - 1)
+    return nodeIsHealthy(endpoint, tries - 1)
   }
 }
 
-/** Gets new endpoints from a user's secondaries */
-const getNewPrimary = async (libs, secondaries) => {
+const getUpToDateSecondary = async (libs, secondaries) => {
   for (const secondary of secondaries) {
     const { isBehind } = await libs.creatorNode.getSyncStatus(secondary)
     if (!isBehind) {
       return secondary
     }
   }
-  throw new Error(`Could not find valid secondaries for user ${secondaries}`)
+  return null
 }
 
+/** If selected primary or secondaries unhealthy, select new one and update user state. */
 const rolloverNodes = async (libs, creatorNodeWhitelist) => {
```

Comment:
in this function, the main changes are to add secondary failover. previously it would only `checkPrimaryHealthy()` on line 37 and then replace with secondary and backfill secondary. but now even if primary healthy, we can also rotate out an unhealthy secondary.